### PR TITLE
Add Alma as a payment method into payment sheet

### PR DESCRIPTION
**Pull request by lisaliu-stripe**
## Summary

Added Alma as a new payment method type

## Motivation

[THREEP_API-1035](https://jira.corp.stripe.com/browse/THREEP_API-1035)

## Testing

Added tests and manually tested

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift [hunk 1]
```diff
@@ -51,6 +52,8 @@ final class PaymentSheet_LPM_ConfirmFlowTests: XCTestCase {
                 return STPTestingJPPublishableKey
             case .BR:
                 return STPTestingBRPublishableKey
+            case .FR:
+                return STPTestingFRPublishableKey
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:56]: We should add a test for alma here, too.

Comment (lisaliu-stripe) [reply StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:56]: Yeah that was the plan but alma test mode currently routes to alma's backend, which is not fully working. You can see from this [payment authorization upstream conversation](https://admin.corp.stripe.com/upstream-conversation/payauthz_QlFkb2l4UENDT0dTOTBJeW41Rk1NTEstU0JrNVJkejMybXd3N285dHVIQSMSFwoVYWNjdF8xSnRnZlFLRzZ2YzdyN1lD). Therefore currently the confirm workflow doesn't really work.

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift [hunk 2]
```diff
@@ -171,6 +171,16 @@ final class PaymentSheet_LPM_ConfirmFlowTests: XCTestCase {
             XCTAssertEqual(form.getAllSubElements().count, 1)
         }
     }
+    
+    func testAlmaConfirmFlows() async throws {
+        try await _testConfirm(intentKinds: [.paymentIntent],
+                               currency: "EUR",
+                               paymentMethodType: .dynamic("alma"),
+                               merchantCountry: .FR) { form in
+            // AmazonPay has no input fields
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:180]:
```suggestion
            // Alma has no input fields
```

**Review by yuki-stripe — APPROVED** [review]
Run ` ./ci_scripts/format_modified_files.sh` to appease the linter

File: StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift [hunk 3]
```diff
@@ -169,6 +172,16 @@ final class PaymentSheet_LPM_ConfirmFlowTests: XCTestCase {
         }
     }
 
+    func testAlmaConfirmFlows() async throws {
+        try await _testConfirm(intentKinds: [.paymentIntent],
+                               currency: "EUR",
+                               paymentMethodType: .dynamic("alma"),
+                               merchantCountry: .FR) { form in
+            // AmazonPay has no input fields
```
Comment (yuki-stripe) [inline StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/PaymentSheetLPMConfirmFlowTests.swift:180]: This still says AmazonPay :P

**Review by yuki-stripe — APPROVED** [review]

**Pull request merged**